### PR TITLE
UI build

### PR DESCRIPTION
**Pull request by devendra-alt**
## 🪶 Features Added
✔️ Created an array of the book object
✔️ Added book array to `localstorage`
✔️ `localstorage` books being displayed on loading on the website
✔️ Add button is used for adding a new book to list
✔️ Remove button being used for removing the book from the list
✔️ DOM is used to create and update the book list
✔️ Correct book being added or removed according to user interaction

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  books = getLocalStorageBooks();
+
+  books.push(Book);
+
+  const booksString = JSON.stringify([books, Book.id]);
+  localStorage.setItem('books', booksString); // updating the books array
+
+  createBookListItem();
+});
+
```
Comment (DuaneDave) [inline script.js:90]: Good job setting up the functionalities, however, the add functionality is not working currently and I think you should fix the errors popping up on the console once the button is clicked in order for the functions to be accessed by a user.

[screen-capture (1).webm](https://github.com/tajulafreen/Awesome-Books/assets/84930355/3e1dec72-0beb-46a4-baf2-163d192422b3)

**Review by DuaneDave — CHANGES_REQUESTED** [review]
Hi @devendra-alt, @tajulafreen and @melkamu12 👋, 

Good job so far on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Clean and well written code

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: script.js (hunk 1 above)
Comment (devendra-alt) [reply script.js]: Hello @DuaneDave, 🙇‍♂️ Thank you for the code review, We finally fixed the issue for add button

File: script.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  books = getLocalBooks();
+
+  books.push(Book);
+
+  const booksString = JSON.stringify([books, Book.id]);
+  localStorage.setItem('books', booksString); // updating the books array
+
+  createBookListItem();
+});
```
Comment (aamirkhan2478) [inline script.js:99]: - [ ] As mentioned by the previous reviewer :point_up_2:  please fix the added functionality your add functionality is still not working the following are problems of your project :slightly_smiling_face: 

1. When a user clicks the "Add" button:

   a. A new book is added to the collection.

   b. The new book is displayed on the page.
1. When a user clicks the "Remove" button:

   a. The right book is removed from the collection. 

   b. The right book disappears from the page.
1. Data is preserved in the browser's memory by using [localStorage](https://github.com/microverseinc/curriculum-html-css/blob/main/local_storage.md), reload the page manually to check this after adding a book and removing a book. When a user clicks the "Add" button

_**Check below :point_down: my video for more details**_

[screen-capture (16).webm](https://github.com/tajulafreen/Awesome-Books/assets/45226848/661a7fab-e3ea-4a18-b229-c4a26913dda9)

**Review by aamirkhan2478 — CHANGES_REQUESTED** [review]
Hi @devendra-alt, @tajulafreen & @melkamu12 



Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

 ## Highlights
✔️ Good Effort
✔️ Your design is good.
✔️ All linters passed

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**